account_mtd/models/mtd_hello_world.py

```diff
@@ -0,0 +1,456 @@
+# -*- coding: utf-8 -*-
+
+import requests
+import json
+import logging
+import werkzeug
+import urllib
+
+from odoo import models, fields, api, exceptions
+from datetime import datetime, timedelta
+from urllib.parse import urlparse
+
+
+class MtdHelloWorld(models.Model):
+    _name = 'mtd.hello_world'
+    _description = "Hello world to test connection between Odoo and HMRC"
+    
+    _logger = logging.getLogger(__name__)
+    
+    name = fields.Char('Name', required=True, readonly=True)
+    api_name = fields.Many2one(comodel_name="mtd.api", required=True)
+    hmrc_configuration = fields.Many2one(comodel_name="mtd.hmrc_configuration", string='HMRC Configuration')
+    scope = fields.Char(related="api_name.scope")
+    response_from_hmrc = fields.Text(string="Response From HMRC", readonly=True)
+    which_button_type_clicked = fields.Char(string="which_button")
+    path = fields.Char(string="sandbox_url")
+    current_record = fields.Char()
+    # need a variable which will keep the record id so that when we
+    
+    @api.multi
+    def action_hello_world_connection(self):
+        if not self.hmrc_configuration:
+            raise exceptions.Warning("Please select HMRC configuration before continuing!")
+        # making sure that the endpoint is what we have used in the data file as the name can be changed anytime.
```

Comment:
I'd rather use the ID rather than the names as if they are changed in future then at least this would not cause errors. I have already had to change the name three times and had to remember to change the code accordingly. Wich is why I have opted out for this option.

Reply:
So do they give us an id in some sort of response, then you pass it back to them later?